packages/components/src/components/TokenInput/TokenInput.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  BoxProps,
+  FormControl,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputProps,
+  InputRightElement,
+  Select,
+  SelectFieldProps,
+} from '@chakra-ui/react';
+import React from 'react';
+import { defaultTokens } from './tokens';
+
+export interface Token {
+  // token sybmol
+  symbol: string;
+  // token contract address
+  address: string;
+}
+
+export interface TokenInputProps {
+  /**
+   * an array of tokens to display
+   */
+  tokens?: Token[];
+  /**
+   * value of the input
+   */
+  value: string;
+  /**
+   * onChange handler
+   */
+  onValueChange: (value: string) => void;
+  /**
+   * label for the input
+   */
+  label?: string;
+  /**
+   * selected token
+   */
+  selectedToken: string;
+  /**
+   * token change hanlder
+   */
+  onTokenChange: (token: string) => void;
+  /**
+   * want to show default tokens
+   */
+  showDefualtTokens?: boolean;
```

Comment:
I think we should probably just let users provide their own tokens. There's a lot of bikeshedding that could happen here:

- Why are these tokens included as defaults but not others?
- What if I want all of the default tokens except one of them?

At the very least, these should be documented in the `jsdoc` string so users know what they're going to get by enabling this.